### ISSUE TITLE
SAML authentication

### ISSUE THREAD

**Issue by osdiab** [better-auth/better-auth#96]
For enterprise usecases; it is possible to achieve this with OAuth and running SAML Jackson, but would be nice to not need that extra dependency.

https://github.com/boxyhq/jackson

**Comment by HeyGarrison** [comment]
+1

**Comment by gsaandy** [comment]
Hello, awesome community! Any updates on this?

**Comment by stevica-canadi** [comment]
+1

**Comment by spugachev** [comment]
+1

**Comment by leobudima** [comment]
+1

**Comment by amxv** [comment]
+1. @ssoready also provides a SAML over OAuth flow. They have some code examples for NextAuth.js but I imagine this can't be too hard to implement with Better Auth:

[https://ssoready.com/docs/saml-over-oauth-saml-nextauth-integration](https://ssoready.com/docs/saml-over-oauth-saml-nextauth-integration)

**Comment by WonderPanda** [comment]
This would be an unbelievably massive contribution to the auth ecosystem for NodeJS. We currently use SAML Jackson as an additional service to facilitate a SAML over OAuth Flow but if Better Auth could do this instead and we could drop this additional dependency it would be huge

**Comment by ddwebdevv** [comment]
YES please! we need SAML too! want to use better-auth for everything!

**Comment by SohelIslamImran** [comment]
+1

**Comment by MattLParker** [comment]
+1

**Comment by aheidelberg** [comment]
+1

**Comment by mshornikov** [comment]
+1

**Comment by tedbrine** [comment]
Would be great if a SAML IdP plugin came from this too.

**Comment by infoflowtech** [comment]
+1

**Comment by yassinrais** [comment]
+1

**Comment by jaime-rr** [comment]
+1

**Comment by Korabi123** [comment]
+1

**Comment by yamz8** [comment]
+1

**Comment by jakst** [comment]
Please stop commenting +1. It sends useless notifications to everyone who subscribes to this issue. Add a thumbs up to the original post if you want to show your support.

**Comment by gellaz** [comment]
+1

**Comment by maielo** [comment]
+1

**Comment by thinh-trinh-duc** [comment]
~~+1~~
Damn it, I accidentally posted before read [this](https://github.com/better-auth/better-auth/issues/96#issuecomment-2687671970) comment.
Sorry guys

**Comment by singh05** [comment]
+1

**Comment by Bekacru** [comment]
thanks for upvoting. soon! top of our list :))

**Comment by Qodestackr** [comment]
+2

**Comment by Qodestackr** [comment]
> thanks for upvoting. soon! top of our list :))

Any guide on working with OIDC? I am having alot of pain figuring out, esp lack of jwks endpoint and how to find a way to atleast work. Also, kindly mention on docs Vercel has quirks when reaching `.well-known/openid-configuration`

**Comment by Hyperscale1** [comment]
+1

**Comment by stevica-attensi** [comment]
+1

**Comment by younisshah** [comment]
👍🏻

**Comment by younisshah** [comment]
+1

**Comment by ulic75** [comment]
+1

**Comment by kvnm93** [comment]
+1

**Comment by Hyperscale1** [comment]
+1

**Comment by rinarakaki** [comment]
I’d appreciate it if you could share with us the estimated timeline for SAML support to be released (and the working branch/PR)!  Because this really is the blocker for us to decide to migrate from next-auth.

**Comment by SohelIslamImran** [comment]
> I’d appreciate it if you could share with us the estimated timeline for SAML support to be released (and the working branch/PR)! Because this really is the blocker for us to decide to migrate from next-auth.

Same... @Bekacru

**Comment by chitalian** [comment]
+1

**Comment by uyrabi** [comment]
+1

**Comment by Whats-A-MattR** [comment]
> Any guide on working with OIDC? I am having alot of pain figuring out, esp lack of jwks endpoint and how to find a way to atleast work. Also, kindly mention on docs Vercel has quirks when reaching `.well-known/openid-configuration`

This has nothing to do with the Issue you're commenting on.

**Comment by mags-sully** [comment]
I can't get buy-in on the project without SAML 😭

**Comment by niraj-khatiwada** [comment]
+1

**Comment by mc-eugenea** [comment]
+1

**Comment by SinanOezkan** [comment]
+1

**Comment by standrd** [comment]
+1, been checking this issue frequently, can't wait to start using better-auth once this is in.

**Comment by maccman** [comment]
+1

**Comment by smlarkin** [comment]
+1

**Comment by agoldstein03** [comment]
+1, would influence our decision to switch :)

**Comment by TahaBoulehmi** [comment]
+1

**Comment by JWijnakker** [comment]
+1

**Comment by Dawsson** [comment]
+1

**Comment by yamz8** [comment]
Coming soon fellas 

https://x.com/imbereket/status/1924143626260770975?s=46

**Comment by imranhou** [comment]
+1

**Comment by mfilipiec** [comment]
+1

**Comment by railaru** [comment]
+1

**Comment by aimming** [comment]
+1

**Comment by FredrikZeiner** [comment]
> Coming soon fellas
> 
> https://x.com/imbereket/status/1924143626260770975?s=46

Awesome

**Comment by CharlesZhao27** [comment]
+1

**Comment by bhagyamudgal** [comment]
+1

**Comment by LuciaBeron3456** [comment]
+1

**Comment by thattimc** [comment]
+1

**Comment by Nishchit14** [comment]
@Bekacru is there any ETA on this feature?

**Comment by RSginer** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by ahmadalzoubi31** [comment]
+1

**Comment by lecramr** [comment]
+1

**Comment by houssaineamzil** [comment]
+1

**Comment by florianamette** [comment]
+1

**Comment by Cedric0852** [comment]
+1

**Comment by yamz8** [comment]
Coming on the 27th

https://x.com/imbereket/status/1934473343580975381?s=46

**Comment by blobgetblob** [comment]
+1, also saml-provider would be great (like oidc-provider), for some legacy services thats only supports saml...

**Comment by rajdeep-ghosh** [comment]
+1

**Comment by Gautham495** [comment]
+1

**Comment by niraj-khatiwada** [comment]
Thanks @Bekacru .Can't wait to implement this on my product.

**Comment by Nishchit14** [comment]
@Bekacru The doc is not updated yet or SAML is not released yet in production?  just curious to know.

<img width="859" alt="Image" src="https://github.com/user-attachments/assets/2da0caf9-c5aa-45d4-b48b-c46f35b34044" />

**Comment by mzupan** [comment]
https://beta.better-auth.com/docs/plugins/sso

You also need to install the beta version
